eth/contracts/libraries/LibSpaceship.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+/**
+ * Library for all things Spaceships
+ */
+
+// Contract imports
+import "hardhat/console.sol";
+
+// Library imports
+import {LibUtils} from "./LibUtils.sol";
+
+// Storage imports
+import {LibStorage, GameStorage, GameConstants, SnarkConstants} from "./LibStorage.sol";
+
+// Type imports
+import {Spaceship, SpaceshipInfo, SpaceshipType, TokenType} from "../DFTypes.sol";
+
+library LibSpaceship {
```

Comment:
As discussed in DMs, I think `LibSpaceship` and `LibArtifact` should each have a `create` function that takes arguments to produce a valid struct (instead of creating the structs ad-hoc). This would allow the structs to **always** have a correct `id` instead of creating the ad-hoc structs with an ID of 0 and then fixing them later.

Reply:
Yup, will do.